app/waf/src/controllers/adc.controller.ts

```diff
@@ -32,27 +31,35 @@ export class AdcController {
   @post(prefix + '/adcs', {
     responses: {
       '200': {
-        description: 'Adc model instance',
-        content: {'application/json': {schema: {'x-ts-type': Adc}}},
+        description: 'Successfully create ADC resource',
+        content: {'application/json': Schema.adcResponse},
+      },
+      '400': {
+        description: 'Invalid ADC resource',
+        content: {'application/json': Schema.BadRequest},
+      },
+      '422': {
+        description: 'Unprocessable ADC resource',
+        content: {'application/json': Schema.UnprocessableEntity},
       },
     },
   })
-  async create(@requestBody() adc: Partial<Adc>): Promise<Adc> {
-    if (!adc.id) {
-      adc.id = uuid();
-    }
-
+  async create(
+    @requestBody(Schema.adcCreateRequest) reqBody: Partial<Adc>,
+  ): Promise<AdcResponse> {
     try {
-      return await this.adcRepository.create(adc);
+      return Promise.resolve(
```

Comment:
My fault that, here, returning Promise and value itself are both OK. 
But in catch block, I suggest `throw new HttpErrors.BadRequest(error.message);`

Reply:
Let us change them from Promise.resolve/reject back to 'return' and 'throw'.

Reply:
Done.